### DISCUSSION TITLE
ANN: Alternative Test Tool

### DISCUSSION THREAD

**Discussion by paulushub** [linebender/resvg-test-suite#40] [Show and tell]
For Windows platform, and currently with C#/WPF (Win-Forms version is planned), I have created a tool for manually testing and logging test results.
* It comes with a custom build of the **Svg-Net** console application (again Windows specific).
* It will automatically download the tests from this repository and insert style specifying font families for tools that supports it (using `@font-face`), ensuring a consistent font support.
* It will automatically download released version of **Resvg** and **Batik** libraries as test tools.
* It will automatically detect installed versions of **Inkscape**, **Firefox** and **ImageMagick** (which comes with the latest build of **LibRsvg** for Windows).
* For portable versions of **ImageMagick** or **LibRsvg**, it provides settings page to select the tool location paths.
* It embeds Chronium-based **Edge Browser** as a testing tool (also for testing hyperlinks).
* There are no supports for **wxSVG** and **QtSVG**.
* The test tools are run in out-of-process mode and crashes handled gracefully.

I will be using this to confirm the expected test results and provide suggestions, where necessary.

![WpfTestOtherSvg1](https://user-images.githubusercontent.com/809351/127999665-0a62c28b-ad81-4394-a84f-a0a372c5ea5f.png)
![WpfTestOtherSvg2](https://user-images.githubusercontent.com/809351/127999691-dbe564f5-885c-4659-bf28-2f0dbe28ffe0.png)

**Comment by RazrFalcon** [comment]
Nice! I have basically the same one, but crossplatform: https://github.com/RazrFalcon/resvg-test-suite/tree/master/tools/vdiff bu it's probably not as user-friendly.

**Reply by paulushub** [reply]
>  I have basically the same one, but crossplatform: 

Yes, and so this is an alternative test tool to the **vdiff**.
It is an improvement on W3C SVG Test Suite tool I created, with added support for the multiple-test tools feature of **vdiff**.

**Reply by RazrFalcon** [reply]
Oh, I misunderstood the post.

Is this your project: https://github.com/ElinamLLC/SharpVectors ? Is it Windows only?

By the way, there is yet another SVG library appeared: https://github.com/google/skia/tree/main/modules/svg
I plan to test agains it as soon as I figure out how to build it.

**Reply by paulushub** [reply]
> Is this your project: https://github.com/ElinamLLC/SharpVectors ? Is it Windows only?

Yes, my project and it is Windows only targeting WPF mainly and WinForms C# libraries.

> By the way, there is yet another SVG library appeared: 

Is it the original SVG module in the Skia, or a new project? It seems there is no styles support..

> I plan to test agains it as soon as I figure out how to build it.

Please post any progress on this.

**Reply by RazrFalcon** [reply]
>Is it the original SVG module in the Skia, or a new project?

From what I understood it's based on old, experimental code, but they really improved it lately.

Yeah, not sure about CSS. Also text handling has a surpassingly little code.
